docs/persistent_compilation_cache.md

```diff
@@ -67,8 +67,135 @@ Cloud Storage (GCS) bucket. We recommend the following configuration:
 Assuming that `gs://jax-cache` is the GCS bucket, set `cache-location` as
 follows:
 
-```
+```python
 import jax
 
 jax.config.update("jax_compilation_cache_dir", "gs://jax-cache")
 ```
+
+## How it works
+
+The JAX compilation cache works by hashing a number of parameters to create a signature for a compiled function. These are:
+
+* The computation performed by the function captured by the non-optimized HLO of the JAX function being hashed
+
+* The jaxlib version 
+
+* Relevant XLA compilation flags
+
+* Device configuration captured in general, by the number of devices and the topology of the devices. 
+  Currently for GPUs, the topology only contains a string representation of the GPU name
+
+* Compression algorithm used to compress the compiled executable
+
+* Any custom hooks
+
+When the signature for a function created using the parameters above matches 
+that of a compiled function in the persistent cache the function will not be compiled, 
+but will just be read and deserialized from the persistent cache. 
```

Comment:
This pararaph is redundant, because this is just how a cache works. Can just mention that the cache key consists of these items.